src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -152,7 +152,11 @@ internal abstract partial class AbstractProject : ForegroundThreadAffinitizedObj
 
             if (visualStudioWorkspaceOpt != null)
             {
-                this.EditAndContinueImplOpt = new VsENCRebuildableProjectImpl(this);
+                if (Language == LanguageNames.CSharp || Language == LanguageNames.VisualBasic)
```

Comment:
In general it's nice to avoid this at the core layer and use a feature specific override in their projects, but I'm okay with this one.